src/encord_active/lib/dataset/summary_utils.py

```diff
@@ -41,22 +53,67 @@ def get_median_value_of_2d_array(array: np.ndarray) -> np.ndarray:
     return array[item_index[0][0], :]
 
 
-def get_all_annotation_numbers(project_folder: Path) -> Tuple[int, int]:
+def get_all_annotation_numbers(project_paths: ProjectFileStructure) -> AnnotationStatistics:
     """
-    returns (number of classification label, number of object label)
+    Returns label statistics for both objects and classifications. Does not count nested
+    labels, only counts the immediate labels.
     """
+
+    labels: AnnotationStatistics = AnnotationStatistics()
     classification_label_counter = 0
     object_label_counter = 0
 
-    for label_row in (project_folder / "data").iterdir():
+    project_ontology = json.loads((project_paths.ontology).read_text(encoding="utf-8"))
+    ontology = OntologyStructure.from_dict(project_ontology)
+
+    for object_item in ontology.objects:
+        labels.objects[object_item.name] = 0
+    for classification_item in ontology.classifications:
+        labels.classifications[classification_item.attributes[0].name] = {}
```

Comment:
Probably can be replaced with `setdefault` since we already have a case where we use it.

Reply:
Do you mean like the following:

```suggestion
    for object_item in ontology.objects:
        labels.objects.setdefault(object_item.name, 0)
    for classification_item in ontology.classifications:
        labels.classifications.setdefault(classification_item.attributes[0].name, {})
```

Reply:
@Encord-davids

Reply:
Sorry I wasn't clear, I meant in the main loop where we increase the counters.

Reply:
If I understood correctly, we need to initialize all of them from zero because, in the label_row, there might not be labels for all object/classification types, so in this case we need to report that there is a class X in your ontology but it has no labels. If we do it only iterating through label_rows we cannot know it.

Reply:
Ahh totally makes sense, sorry for the oversight from my side 🙇